gradio/blocks.py

```diff
@@ -1269,7 +1288,14 @@ def postprocess_data(
                     "Number of output components does not match number "
                     f"of values returned from from function {block_fn.name}"
                 ) from err
-            block = self.blocks[output_id]
+
+            try:
+                block = self.blocks[output_id]
+            except KeyError as e:
+                raise InvalidBlockError(
+                    f"Output component with id {output_id} used in {dependency['trigger']}() event not found in this gr.Blocks context. You are allowed to nest gr.Blocks contexts, but there must be a gr.Blocks context that contains all components and events."
```

Comment:
For symmetry with the other message:

```suggestion
                    f"Output component with id {output_id} used in {dependency['trigger']} event not found in this gr.Blocks context. You are allowed to nest gr.Blocks contexts, but there must be a gr.Blocks context that contains all components and events."
```

Reply:
You might also consider using `{...!r}` for `repr` formatting.